### ISSUE TITLE
[pickers] RTLProvider missing from @mui/system but used in x-date-pickers

### ISSUE THREAD

**Issue by metinjakupi** [mui/mui-x#14781]
### Steps to reproduce

I was using the version 7.0.0 and it was working  suddenly  i checked the code and  i see this i added 
import { useRtl } from '@mui/system/RtlProvider';
but before this it was like this 

const theme = useTheme();
  const isRTL = theme.direction === 'rtl';

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
System:
    OS: macOS 14.6.1
  Binaries:
    Node: 20.17.0 - ~/.nvm/versions/node/v20.17.0/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.17.0/bin/npm
    pnpm: Not Found
  Browsers:
    Chrome: 129.0.6668.60
    Edge: Not Found
    Safari: 17.6
  npmPackages:
    @emotion/react: 11.10.8 => 11.10.8 
    @emotion/styled: 11.10.8 => 11.10.8 
    @mui/base:  5.0.0-alpha.127 
    @mui/core-downloads-tracker:  5.16.7 
    @mui/lab: 5.0.0-alpha.128 => 5.0.0-alpha.128 
    @mui/material: 5.12.2 => 5.12.2 
    @mui/styled-engine:  5.16.6 
    @mui/system: 5.12.1 => 5.12.1 
    @mui/types:  7.2.17 
    @mui/utils:  5.16.6 
    @mui/x-data-grid: 6.0.3 => 6.0.3 
    @mui/x-date-pickers: ^6.19.7 => 6.19.7 
    @types/react: 18.2.0 => 18.2.0 
    react: ^18.3.1 => 18.3.1 
    react-dom: ^18.3.1 => 18.3.1 
    typescript: 5.0.4 => 5.0.4 
```
</details>

**Search keywords**: RTL

**Comment by flaviendelangle** [comment]
You are using the version `@mui/material: 5.12.2`
We support version `5.15.14` and above

Upgrading should fix your problem

**Comment by metinjakupi** [comment]
Thanks i will try now, but this is strange cause i didn't update anything i had 7.0.0 from months ago and the same material version

**Comment by flaviendelangle** [comment]
We started using `RtlProvider` in a minor version of `@mui/x-date-pickers`, but the minimal supported version of `@mui/material` had been bumped for v7.0.0.

So my understanding is that before you had an incompatible version of `@mui/material`, but it was not causing any issue because we weren't using any feature that you did not have in your version of `@mui/material`.

**Comment by metinjakupi** [comment]
The problem is cause i am using a theme and that has defined settings and if i update the material mui the whole theme is crashed, so should i downgrade the x-date-picker to 6.9?

**Comment by flaviendelangle** [comment]
It's weird that you cannot update `@mui/material` since it's just a minor (if you tried to update to their last v5 release and not the v6 release which includes a handfull of breaking changes)
If you have a problem, don't hesitate to create an issue on their repo.
Downgrading `@mui/x-date-pickers` is of course a solution but not the best one for you.

**Comment by metinjakupi** [comment]
I will downgrade for now cause it will cause me bigger problems, their palette stuff also is old

Error: MUI: Unsupported `47, 43, 61` color.
The following formats are supported: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color().

**Comment by LukasTy** [comment]
@metinjakupi could you provide a minimal reproduction example to see your problem in action?
A live example would be perfect. [This guide](https://mui.com/x/introduction/support/#bug-reproductions) should be a good starting point. Thank you!

**Comment by metinjakupi** [comment]
@LukasTy There are a lots of custom confifgurations there are overrides of react-mui inputs with the themes one, so i think it is better to downgrade :/

**Comment by LukasTy** [comment]
> There are a lots of custom confifgurations there are overrides of react-mui inputs with the themes one, so i think it is better to downgrade :/

Whatever works best for you. 👌 
But you will probably have to bump the versions sooner or later. Hence, I would try to spend some effort to unblock the `@mui/material` bump sooner rather than later. 🤔

**Comment by metinjakupi** [comment]
We can be in touch, i  would like to upgrade to mui 5.15

**Comment by LukasTy** [comment]
> We can be in touch, i would like to upgrade to mui 5.15

We would like not to keep issues that have nothing actionable open.
I'm closing it as the root cause of the issue seems clear.
If you still experience issues upgrading to the latest v5 of `@mui/material`—feel free to open an issue on their repository if the problem is with that library. 👌 😉

**Comment by caelus-dev** [comment]
Hi, I have similar problem. The case is different, but the error is similar, so I didn't open a new ticket.

I used @mui/material: `"^5.16.7"` and @mui/x-date-pickers: `"^7.21.0"`.
I wanted to upgrade material ui to v6, but I got the following errors:

```
[ERROR] Could not resolve "@mui/system/RtlProvider"
[ERROR] Could not resolve "@mui/system"
```

These errors are from mui/x-date-pickers,

In the official documentation the only thing is mentioned in the v5 to v6 migration section about date picker is:
> MUI X packages do not follow the same versioning strategy as Material UI. If you're using any of the following packages, they should remain unchanged during the upgrade process

So I don't understand. Can't the mui/x-date-pickers be used with v6 material ui? Or what's the problem?

**Comment by flaviendelangle** [comment]
Hi,
Could you indicate to me which version of `@mui/system` ended up being installed when you had the error?

**Comment by caelus-dev** [comment]
> Hi, Could you indicate to me which version of `@mui/system` ended up being installed when you had the error? It would be even better if you could create a new issue to have a clean start 👍

`@mui/system` is totally missing and I didn't put it into the packages, because in the installation guide there is no mention about it, or maybe I missed something.

If my answer doesn't resolve the problem and more details are necessary, I will open a new ticket.

**Comment by flaviendelangle** [comment]
Please open a new ticket :+1: 
You should have `@mui/system` installed automatically `@mui/material` but we'll find what's wrong

**Comment by caelus-dev** [comment]
> Please open a new ticket 👍 You should have `@mui/system` installed automatically `@mui/material` but we'll find what's wrong

Thank you, I opened a ticket for the problem: #15267
